src/Evolution/Systems/Cce/Initialize/NoIncomingRadiation.cpp

```diff
@@ -0,0 +1,168 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/Cce/Initialize/NoIncomingRadiation.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "DataStructures/SpinWeighted.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/Cce/Initialize/InitializeJ.hpp"
+#include "NumericalAlgorithms/OdeIntegration/OdeIntegration.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/Printf.hpp"
+#include "Time/History.hpp"
+#include "Time/Slab.hpp"
+#include "Time/Time.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Time/TimeSteppers/RungeKutta3.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeString.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Cce {
+namespace InitializeJ {
+namespace detail {
+
+void radial_evolve_psi0_condition(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, 2>*> volume_j_id,
+    const SpinWeighted<ComplexDataVector, 2>& boundary_j,
+    const SpinWeighted<ComplexDataVector, 2>& boundary_dr_j,
+    const SpinWeighted<ComplexDataVector, 0>& r, const double radial_step,
```

Comment:
why do you need to specify either the `radial_step` or the `number_of_radial_points`?  Aren't you integrating out all the way to scri and only need the values there; or are we confused about something?

Reply:
You can estimate the step size using the algorithm `Starting Step Size` (page 169) of II.4 Practical Error Estimation and Step Size Selection of Solving Ordinary Differential Equations by Hairer, Norsett, Wanner.  Probably h_0 of part b of the algorithm is enough.